nanoFramework.Hardware.Esp32.Rmt/RmtChannelSettings.cs

```diff
@@ -0,0 +1,142 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System;
+
+namespace nanoFramework.Hardware.Esp32.Rmt
+{
+    /// <summary>
+    /// Base class for shared RMT Channel settings.
+    /// </summary>
+    public abstract class RmtChannelSettings
+    {
+        private int _channel;
+        private int _pinNumber;
+        private byte _clockDivider;
+        private byte _numberOfMemoryBlocks;
+        private int _bufferSize;
+        private bool _signalInverterEnabled;
+
+        /// <summary>
+        /// Gets or sets the channel number.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public int Channel
+        {
+            get => _channel;
+            set => _channel = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the GPIO pin number to be used with the specified channel.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public int PinNumber
+        {
+            get => _pinNumber;
+            set => _pinNumber = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the clock divider value.
+        /// </summary>
+        /// <remarks>
+        /// ESP IDF v4.4.3 only supports the APB bus clock @ 80Mhz.
+        /// The default value for the divider is 80 (80Mhz / 80 = 1Mhz = 1us ticks).
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">Cannot make changes to the property after the RMT channel is initialized.
+        /// Use the methods on the channel instance to make configuration changes.</exception>
+        public byte ClockDivider
+        {
+            get => _clockDivider;
+            set => _clockDivider = value;
```

Comment:
similar question as for the rest, any range value?

Reply:
Acceptable range is 1 to 255. I've updated this part.